src/m365/purview/commands/retentionevent/retentionevent-add.ts

```diff
@@ -0,0 +1,174 @@
+        }
+
+        if (!args.options.assetIds && !args.options.keywords) {
+          return 'Specify assetIds and/or keywords, but at least one.';
+        }
+
+        return true;
+      }
+    );
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        description: typeof args.options.description !== 'undefined',
+        triggerDateTime: typeof args.options.triggerDateTime !== 'undefined',
+        assetIds: typeof args.options.assetIds !== 'undefined',
+        keywords: typeof args.options.keywords !== 'undefined'
+      });
+    });
+  }
+
+  #initOptionSets(): void {
+    this.optionSets.push(
+      { options: ['eventTypeId', 'eventTypeName'] }
+    );
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    if (accessToken.isAppOnlyAccessToken(auth.service.accessTokens[this.resource].accessToken)) {
+      this.handleError('This command does not support application permissions.');
+    }
+
+    if (this.verbose) {
+      logger.logToStderr(`Creating retention event...`);
+    }
+
+    const eventQueries: any[] = [];
+
+    if (args.options.assetIds) {
+      eventQueries.push({ queryType: "files", query: args.options.assetIds });
+    }
+
+    if (args.options.keywords) {
+      eventQueries.push({ queryType: "messages", query: args.options.keywords });
+    }
+
+    const eventTypeId = await this.getEventTypeId(logger, args);
+
+    const data = {
+      'retentionEventType@odata.bind': `https://graph.microsoft.com/beta/security/triggerTypes/retentionEventTypes/${eventTypeId}`,
+      displayName: args.options.displayName,
+      description: args.options.description,
+      eventQueries: eventQueries,
+      eventTriggerDateTime: args.options.triggerDateTime
+    };
+
+    try {
+      const requestOptions: CliRequestOptions = {
+        url: `${this.resource}/beta/security/triggers/retentionEvents`,
+        headers: {
+          accept: 'application/json;odata.metadata=none'
+        },
+        responseType: 'json',
+        data: data
+      };
+
+      const res: any = await request.post<any>(requestOptions);
+      logger.log(res);
+    }
+    catch (err: any) {
+      this.handleRejectedODataJsonPromise(err);
+    }
+  }
+
+  private async getEventTypeId(logger: Logger, args: CommandArgs): Promise<string> {
+    if (args.options.eventTypeId) {
+      return args.options.eventTypeId;
+    }
+
+    if (this.verbose) {
+      logger.logToStderr(`Retrieving the event type id for event type ${args.options.eventTypeName}`);
+    }
+
+    const items: any = await odata.getAllItems(`${this.resource}/beta/security/triggers/retentionEvents`);
+
+    const eventTypes = items.filter((x: any) => x.displayName === args.options.eventTypeName);
+
+    if (eventTypes.length === 0) {
+      throw `The specified event type '${args.options.eventTypeName}' does not exist.`;
+    }
+
+    if (eventTypes.length > 1) {
```

Comment:
I think I commented on this before; you can remove this check as `eventTypeName` is a unique value, there cannot be more than one with tje same name.